modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OIDCDiscoveryTestCase.java

```diff
@@ -30,57 +33,59 @@
 import org.testng.annotations.AfterClass;
 import org.testng.annotations.BeforeClass;
 import org.testng.annotations.DataProvider;
-import org.testng.annotations.Factory;
 import org.testng.annotations.Test;
 import org.wso2.charon.core.schema.SCIMConstants;
 import org.wso2.identity.integration.common.utils.ISIntegrationTest;
 
 import java.io.IOException;
 
+import static org.testng.Assert.assertEquals;
+
 public class OIDCDiscoveryTestCase extends ISIntegrationTest {
 
     public static final String WEBFINGER_ENDPOINT_SUFFIX = "/.well-known/webfinger";
     public static final String RESOURCE = "resource";
     public static final String REL = "rel";
     private String isServerBackendUrl;
-    private String webfingerEndpoint;
-    private String relUri = "http://openid.net/specs/connect/1.0/issuer";
-    private String discoveryBasePath;
-    private DiscoveryConfig config;
-
-    @Factory(dataProvider = "webfingerConfigProvider")
-    public OIDCDiscoveryTestCase(DiscoveryConfig config) {
-        if (log.isDebugEnabled()){
-            log.info("SAML SSO Test initialized for " + config);
-        }
-        this.config = config;
-    }
 
+    private static final String[] expectedResponseModes = {"fragment", "jwt", "fragment.jwt", "query", "form_post",
+            "query.jwt", "form_post.jwt"};
+    private static final String[] expectedAuthModes = {"private_key_jwt", "client_secret_post", "tls_client_auth",
+            "client_secret_basic"};
+    private static final String[] expectedResponseTypes = {"id_token token", "code", "code id_token token",
+            "code id_token", "id_token", "code token", "none", "device", "subject_token", "id_token subject_token",
+            "token"};
+    private static final String[] expectedGrantTypes = {"refresh_token", "password", "client_credentials", "iwa:ntlm",
+            "urn:ietf:params:oauth:grant-type:saml2-bearer", "urn:ietf:params:oauth:grant-type:device_code",
+            "authorization_code", "account_switch", "urn:ietf:params:oauth:grant-type:token-exchange",
+            "organization_switch", "urn:ietf:params:oauth:grant-type:jwt-bearer"};
 
     @BeforeClass(alwaysRun = true)
     public void testInit() throws Exception {
         super.init();
         isServerBackendUrl = isServer.getContextUrls().getWebAppURLHttps();
-        webfingerEndpoint = isServerBackendUrl + WEBFINGER_ENDPOINT_SUFFIX + "?" + RESOURCE + "=" + config
-                .getResource() + "&" + REL + "=" + relUri;
     }
 
     @AfterClass(alwaysRun = true)
     public void atEnd() {
 
     }
 
-    @Test(alwaysRun = true, groups = "wso2.is", description = "webfinger test")
-    public void testWebFinger() throws IOException {
+    @Test(alwaysRun = true, groups = "wso2.is", description = "webfinger test",
+            dataProvider = "webfingerConfigProvider")
+    public void testWebFinger(DiscoveryConfig config) {
 
+        String relUri = "http://openid.net/specs/connect/1.0/issuer";
```

Comment:
Shall we give a meaningful name?

Reply:
I think this is meaningful enough. The value is called `rel`

Reply:
relURI -> relationship url.

The rel parameter allows clients to specify what type of information they are seeking about the subject. here we are looking for an OpenID Connect issuer for the given subject.